tests/integration/tests/test_smoke.py

```diff
@@ -2,18 +2,52 @@
 # Copyright 2024 Canonical, Ltd.
 #
 import logging
+from typing import List
 
-from test_util import harness
+import pytest
+from test_util import config, harness
 
 LOG = logging.getLogger(__name__)
 
 
-def test_smoke(session_instance: harness.Instance):
+@pytest.mark.node_count(1)
+@pytest.mark.disable_k8s_bootstrapping()
+def test_smoke(instances: List[harness.Instance]):
+    instance = instances[0]
+
+    bootstrap_smoke_config_path = "/home/ubuntu/bootstrap-smoke.yaml"
+    instance.send_file(
+        (config.MANIFESTS_DIR / "bootstrap-smoke.yaml").as_posix(),
+        bootstrap_smoke_config_path,
+    )
+
     # Verify the functionality of the k8s config command during the smoke test.
     # It would be excessive to deploy a cluster solely for this purpose.
-    result = session_instance.exec(
+    result = instance.exec(
         "k8s config --server 192.168.210.41".split(), capture_output=True
     )
-    config = result.stdout.decode()
-    assert len(config) > 0
-    assert "server: https://192.168.210.41" in config
+    kubeconfig = result.stdout.decode()
+    assert len(kubeconfig) > 0
+    assert "server: https://192.168.210.41" in kubeconfig
+
+    # Verify extra node configs
+    content = instance.exec(
+        ["cat", "/var/snap/k8s/common/args/conf.d/bootstrap-extra-file.yaml"],
+        capture_output=True,
+    )
+    assert content.stdout.decode() == "extra-args-test-file-content"
+
+    # For each service, verify that the extra arg was written to the args file.
+    for service, value in {
+        "kube-apiserver": "--request-timeout=2m",
+        "kube-controller-manager": "--leader-elect-retry-period=3s",
+        "kube-scheduler": "--authorization-webhook-cache-authorized-ttl=11s",
+        "kube-proxy": "--config-sync-period=14m",
+        "kubelet": "--authentication-token-webhook-cache-ttl=3m",
+        "containerd": "--log-level=debug",
+        "k8s-dqlite": "--watch-storage-available-size-interval=6s",
+    }.items():
```

Comment:
hmm, as discussed earlier, this is a unit test in disguise. This is important because we would not like this test to break if any of the flags here have a side-effect on the respective service.

But I see why this is here given the situation.

Reply:
I partially agree. However, we currently don't have a good way to unittest the logic in e.g. `hook_bootstrap.go`. I much rather have this unittest in disguise than silently breaking the extra-args feature which can go unnoticed for quite a while. 
I don't think the changed arguments will affect this test (**famous last words...**). All of them are very close to the default and only differ by a second.